syntax = "proto3";

import "protos/Article/Article.proto";

message GetArticlesRequest {
    int32 client_id = 1;

    // article info:
    string type = 2;
    string author = 3;
    Date date = 4;   
}

message GetArticlesResponse {
    repeated Article article_list = 1;
}

message PublishArticleRequest {
    int32 client_id = 1;
    Article article = 2;
}

message PublishArticleResponse {
    enum Status {
        FAILED = 0;
        SUCCESS = 1;
    }

    Status status = 1;
}

service ClientServer {
    rpc GetArticles (GetArticlesRequest) returns (GetArticlesResponse);
    rpc PublishArticle (PublishArticleRequest) returns (PublishArticleResponse);
}
